services/vision/mlvision/detector.go

```diff
@@ -115,18 +120,42 @@ func attemptToBuildDetector(mlm mlmodel.Service, inNameMap, outNameMap *sync.Map
 		if err != nil {
 			return nil, err
 		}
-		categories, err := convertToFloat64Slice(outMap[categoryName].Data())
-		if err != nil {
-			return nil, err
-		}
 		scores, err := convertToFloat64Slice(outMap[scoreName].Data())
 		if err != nil {
 			return nil, err
 		}
+		hasCategoryTensor := false
+		categories := make([]float64, len(scores)) // default 0 category if no category output
+		if categoryName != "" {
+			hasCategoryTensor = true
+			categories, err = convertToFloat64Slice(outMap[categoryName].Data())
+			if err != nil {
+				return nil, err
+			}
+		}
+		// sometimes categories are stuffed into the score output. separate them out.
+		if !hasCategoryTensor {
+			shape := outMap[scoreName].Shape()
+			if len(shape) == 3 { // cartegories are stored in 3rd dimension
+				nCategories := shape[2]              // nCategories usually in 3rd dim, but sometimes in 2nd
+				if 4*nCategories == len(locations) { // it's actually in 2nd dim
```

Comment:
Helpful comments. In general, I'm slightly worried about the criteria for knowing when the categories are just unreported and you need to look in the scores. We may have to come back to it but that feels like tomorrow's problem.